config/src/main/java/io/confluent/common/config/ConfigDef.java

```diff
@@ -68,9 +68,17 @@
  */
 public class ConfigDef {
 
-  private static final Object NO_DEFAULT_VALUE = new Object();
+  protected static final Object NO_DEFAULT_VALUE = new Object();
 
-  private final Map<String, ConfigKey> configKeys = new HashMap<String, ConfigKey>();
+  private final Map<String, ConfigKey> configKeys;
+
+  public ConfigDef() {
+    configKeys = new HashMap<>();
+  }
+
+  public ConfigDef(ConfigDef baseConfigDef) {
```

Comment:
i wouldn't use the word `base` due to the potential confusion with inheritance. I'd call the parameter `configDef`.